unordered_associative_task_2/student/src/impl.cpp

```diff
@@ -25,11 +25,68 @@ auto xx_hash = [] (const std::string &file)
     return xxh::xxhash<64>(buffer);
 };
 
+/** @todo With help of std filesystem, lists all regular files under specified directory recursively*/
+std::vector<std::string> listFiles(const std::string& directory)
+{
+    std::vector<std::string> fileList;
+
+    for (auto& it : fs::recursive_directory_iterator(directory))
+    {
+        if (it.path().has_extension())
+        {
+            fileList.push_back(it.path().string());
+        }
+    }
+
+    return fileList;
+}
+
+/** @todo Implement function that will remove group in if it has only one or zero elements */
+void removeUniqueGroups(filtering_map& filteredData)
+{
+    for (auto it = filteredData.begin(); it != filteredData.end();)
+    {
+        if (it->second.size() <= 1)
+        {
+            it = filteredData.erase(it);
+        }
+        else
+        {
+            it++;
+        }
+    }
+}
+
+/** @todo Implement function that will transform map to a vector */
+std::vector<std::vector<std::string>> flattenGrouped (const filtering_map& grouped)
+{
+    std::vector<std::vector<std::string>> flatGroups;
+
+    std::vector<std::string> group;
+
+    for (auto& it : grouped)
+    {
+        group.assign(it.second.begin(), it.second.end());
+        flatGroups.push_back(group);
```

Comment:
why can't you just use .insert() with two iterators, but rather assign to a temp vector?